gapic/schema/wrappers.py

```diff
@@ -767,6 +767,23 @@ def http_opt(self) -> Optional[Dict[str, str]]:
         # TODO(yon-mg): enums for http verbs?
         return answer
 
+    @property
+    def path_params(self) -> Sequence[str]:
+        if self.http_opt is None:
+            return []
+        pattern = r'\{\w+\}'
+        return [x[1:-1] for x in re.findall(pattern, self.http_opt['url'])]
+
+    @property
+    def query_params(self) -> Set[str]:
+        if self.http_opt is None:
+            return set()
+        body = []
+        if 'body' in self.http_opt.keys():
+            body.append(self.http_opt['body'])
+        all_params = self.input.fields.keys()
+        return set(all_params) ^ set(body + list(self.path_params))
```

Comment:
I agree you're right and I'll change it but there shouldn't be a difference between difference and symmetric difference here since all `body + path_params` should implicitly be included in all_params. I suppose that's why it hasn't failed for me but I'm not sure why I didn't just use `-` in the first place.

Reply:
I have no strong feeling for whether we want difference or XOR; Victor and the spec should take priority there. +1 for adding a test that explicitly checks for the one that we actually want and NOT the other one.

Side note: IMHO the set operators are not very clear, and especially if doing more complex set operations it's more reader friendly to use the named methods.

Reply:
Replying to Victor's ancestral comment: the generated unit tests and 100% coverage thereof is intended to fill a similar goal as golden tests without making people tear their hair out from frustration. Occasionally it is helpful to diff surfaces, though; I like to either stash changes or do sub-branches, invoke the generator on showcase to save the resulting GAPIC, then ediff the relevant files in the two outputs. It can be a little fiddly, but it's *_far_* easier to do as a tactical move than to have golden tests.

Reply:
For reviews, it's _very_ helpful to be able to diff the generated surfaces. That's the main thing I miss about golden files. The frustration, I can do without. ;-)

Reply:
For this particular comment, as Yon and I discussed yesterday: `body + list(self.path_params)` should be a subset of `all_params`, so difference and xor should behave the same. However, difference expresses the intent better and it makes it more likely we can catch or prevent errors if this assumption is violated due to a bug (now or in the future).

Reply:
Excellent. Glad to hear that the issue has been resolved.